src/solver/conv_winoRxS_f3x2.cpp

```diff
@@ -239,141 +120,339 @@ bool ConvBinWinogradRxSf3x2::IsApplicable(const ConvolutionContext& params) cons
-                                 params.n_inputs,  // C
-                                 params.n_outputs, // K
-                                 params.in_height, // HxW
-                                 params.in_width,
-                                 params.out_height, // OHxOW
-                                 params.out_width,
-                                 params.batch_sz, // N
-                                 params,
-                                 false,
-                                 2);
+    const auto n_inputs_per_group  = params.n_inputs / params.group_counts,
+               n_outputs_per_group = params.n_outputs / params.group_counts;
+
+    if(params.direction.IsBackwardWrW())
+    {
+        return IsShaderContraintsMet(params.in_height,
+                                     params.in_width,
+                                     params.kernel_dilation_h,
+                                     params.kernel_dilation_w,
+                                     params.batch_sz,    // N
+                                     n_inputs_per_group, // K
+                                     params.out_height,
+                                     params.out_width,
+                                     params.kernel_size_h,
+                                     params.kernel_size_w,
+                                     n_outputs_per_group, // C
+                                     params);
+    }
+    else
+    {
+        return IsShaderContraintsMet(params.kernel_size_h, // RxS
+                                     params.kernel_size_w,
+                                     params.kernel_stride_h,
+                                     params.kernel_stride_w,
+                                     n_inputs_per_group,  // C
+                                     n_outputs_per_group, // K
+                                     params.in_height,    // HxW
+                                     params.in_width,
+                                     params.out_height, // OHxOW
+                                     params.out_width,
+                                     params.batch_sz, // N
+                                     params);
+    }
 }
 
+/// \todo Consider re-using code from RxS_f2x3.
 ConvSolution ConvBinWinogradRxSf3x2::GetSolution(const ConvolutionContext& params) const
 {
+    // NOLINTNEXTLINE (cppcoreguidelines-avoid-non-const-global-variables)
+    static bool IsWarned;
+    if(!IsWarned)
+    {
+        if(params.GetStream().GetMaxHardwareComputeUnits() > MAX_CU_LIMIT)
+            MIOPEN_LOG_WE(SolverDbId(*this) << ": GPU has "
+                                            << params.GetStream().GetMaxHardwareComputeUnits()
+                                            << "CUs, but this solver supports max "
+                                            << MAX_CU_LIMIT
+                                            << "and thus may show sub-optimal performance.");
+        IsWarned = true;
+    }
```

Comment:
- Do you want to issue this message only once?
- Why would you like to issue an error message for DEV builds?
  - If this is an error, then maybe we shall make it non-applicable?

Reply:
Shader still will work, but there might be severe performance penalties. It's better to set as non-applicable in this case.
Thank you, fixed in 9db3bc6

Reply:
Good!